Sources/_MatchingEngine/Regex/Parse/LexicalAnalysis.swift

```diff
@@ -738,6 +743,143 @@ extension Source {
     }
   }
 
+  /// Consume a PCRE version number.
+  ///
+  ///     PCREVersionNumber -> <Int>.<Int>
+  ///
+  private mutating func expectPCREVersionNumber(
+  ) throws -> AST.Conditional.Condition.PCREVersionNumber {
+    let nums = try recordLoc { src -> (major: Int, minor: Int) in
+      let major = try src.expectNumber().value
+      try src.expect(".")
+      let minor = try src.expectNumber().value
+      return (major, minor)
+    }
+    return .init(major: nums.value.major, minor: nums.value.minor,
+                 nums.location)
+  }
+
+  /// Consume a PCRE version check suffix.
+  ///
+  ///     PCREVersionCheck -> '>'? '=' PCREVersionNumber
+  ///
+  private mutating func expectPCREVersionCheck(
+  ) throws -> AST.Conditional.Condition.Kind {
+    typealias Kind = AST.Conditional.Condition.PCREVersionCheck.Kind
+    let kind = try recordLoc { src -> Kind in
+      let greaterThan = src.tryEat(">")
+      try src.expect("=")
+      return greaterThan ? .greaterThanOrEqual : .equal
+    }
+    return .pcreVersionCheck(.init(kind, try expectPCREVersionNumber()))
+  }
```

Comment:
It's nice to parse these. What do you think we should do, maybe give a warning or error out? It's good to have a targeted diagnostic though.

Reply:
Yeah I think an error would make sense, at least if there's no sensible behavior to give it at runtime. Would we want the parser to emit that, or would it be left up to the AST consumer? (could we have consumers that might want to deal with this node differently?)